claim-check/code-samples/sample-4/ClientProducer4/SampleKafkaProducer.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Azure.Identity;
+using Confluent.Kafka;
+using Microsoft.Extensions.Configuration;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace Pnp.Samples.ClaimCheckPattern
+{
+    /// <summary>
+    /// Implements the Kafka producer to send messages to a Kafka-enabled Event Hub namespace
+    /// Uses Azure Entra ID to authenticate.
+    /// </summary>
+    public class SampleKafkaProducer(IConfiguration configuration)
+    {
+        readonly string _eventHubEndpoint = configuration.GetSection("AppSettings:eventHubEndpoint").Value!;
+        readonly string _kafkaFqdn = configuration.GetSection("AppSettings:eventHubFqdn").Value!;
+        readonly string _topic = configuration.GetSection("AppSettings:eventHubName").Value!;
+
+        public async Task SendMessage(string message)
+        {
+            var userPrincipal = GetCurrentUserPrincipal();
+            try
+            {
+                var config = new ProducerConfig
+                {
+                    BootstrapServers = _kafkaFqdn,
+                    SecurityProtocol = SecurityProtocol.SaslSsl,
+                    SaslMechanism = SaslMechanism.OAuthBearer,
+                    //Debug = "security,broker,protocol", //Uncomment for librdkafka debugging information
+                };
+
+                using var producer = new ProducerBuilder<string, string>(config)
+                    .SetOAuthBearerTokenRefreshHandler(async (client, cfg) =>
+                    {
+                        var credential = new DefaultAzureCredential();
+                        var accessToken = await credential.GetTokenAsync(new Azure.Core.TokenRequestContext([_eventHubEndpoint]));
+                        client.OAuthBearerSetToken(accessToken.Token, accessToken.ExpiresOn.ToUnixTimeMilliseconds(), userPrincipal);
```

Comment:
I know, in practice, it's not going to matter, but any chance you can make `GetCurrentUserPrincipal` take a credential, specifically this one, and extract what is needed instead of creating two of them (line 20)